students/g_rama/lesson04/mailroom_v2.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+def display_donors():
+    """Display donors if user enters list"""
+    for _ in donors_data.keys():
+        print('{}'.format(_))
+
+
+def thank_you():
+    """Send a thank you letter"""
+    fullname = input("Enter the Donor name:\n")
+    if fullname == str("list"):
+        display_donors()
+    elif fullname in donors_data.keys():
+        amount = int(input("Enter the amount of donation:\n"))
+        donors_data[fullname].append(amount)
+        thank_you_letter(fullname, amount)
+    else:
+        amount = int(input("Enter the amount of donation:\n"))
+        donors_data.update({fullname: [amount]})
+        thank_you_letter(fullname, amount)
+
+
+def thank_you_letter(donor_name,amount):
+    """Send a thank you letter"""
+    print(f"Thank you {donor_name} for donating {amount} dollars generously.")
+
+
+def donor_details(**data):
+    """Print the Donor table"""
+    print(f'{"Donor Name":<20} |{"Total Given":>20} |{"Num Gifts":<20} |{"Average Gift":>20.4}\n')
+    print('-'*90)
+    for row in data:
+        print(f'{row:<20} ${sum(data[row]):>20} {len(data[row]):>20} ${(sum(data[row])/len(data[row])):>10.4}\n')
+
+
+def create_report():
+    """Call the function to create the donor details"""
+    data = donors_data
+    donor_details(**data)
+
+
+def send_letters():
+    """Call the function to create the donor details"""
+    data = donors_data
+    send_letters_all(**data)
+
+
+def send_letters_all(**data):
+    for _ in data:
```

Comment:
`_` is only used for variables that are not actually used but you do use it here.; use a real variable please.